### DISCUSSION TITLE
CozoDB query doubts

### DISCUSSION THREAD

**Discussion by shivarajmulimani** [cozodb/cozo#53] [Q&A] [answered]
I have stored a user table which has some 100 records in it.
where the attribute names are username, id, age, height, weight

I want update specific record in the table, 
Basically I want to update the age of a existing username record in a user table, how can I achieve this?

Can you please provide some example?

**Comment by zh217** [comment, marked as answer]
Basically you just `put` a new row, overwriting the existing one. There is no `update` equivalent in CozoDB.

For example, if you want to update based on the ID, you have to retrieve the other fields manually (for the moment):

```
?[username, id, age, height, weight] := id = $id, *user{id, username, height, weight}, age = $new_age

:put user{id => username, age, height, weight}
```

Note that `age` comes from the query parameter, not from the table.

**Reply by infogulch** [reply]
Would it be reasonable for cozo to include an update command that desugars to this?

**Reply by zh217** [reply]
No, desugaring won't make sense except in the simplest of cases, but a real `update` is probably going to be included (no timeline though).
